bodhi/server/models.py

```diff
@@ -554,6 +555,54 @@ def get_pkg_pushers(self, branch, settings):
 
         return (committers, watchers), (committergroups, watchergroups)
 
+    def get_pkg_committers_from_pagure(self, settings):
+        """
+        Pull users who can commit and are watching a package in Pagure.
+
+        Return two two-tuples of lists:
+        * The first tuple is for usernames.  The second tuple is for groups.
+        * The first list of the tuple is for committers. The second is for
+          watchers.
+        """
+        pagure_url = settings.get(
+            'pagure_url', 'https://src.fedoraproject.org/pagure/')
+        # Pagure uses plural names for its namespaces such as "rpms"
+        namespace = self.type.name + 's'
+        package_pagure_url = '{0}/api/0/{1}/{2}'.format(
+            pagure_url.rstrip('/'), namespace, self.name)
+        package_json = pagure_api_get(package_pagure_url)
+
+        committers = set()
+        for access_type in ['owner', 'admin', 'commit']:
+            committers = committers | set(
+                package_json['access_users'][access_type])
+        for access_type in ['admin', 'commit']:
+            # We resolve the group membership here since these are Pagure
+            # groups, and not FAS groups. Bodhi understands the latter but not
+            # the former.
+            for group_name in package_json['access_groups'][access_type]:
+                group_pagure_url = '{0}/api/0/group/{1}'.format(
+                    settings['pagure_url'].rstrip('/'), group_name)
+                group_json = pagure_api_get(group_pagure_url, timeout=30)
+                committers = committers | set(group_json['members'])
+
+        watchers = set()
+        package_watchers_pagure_url = '{0}/watchers'.format(package_pagure_url)
+        watchers_json = pagure_api_get(package_watchers_pagure_url)
+        for watcher, watcher_levels in watchers_json['watchers'].items():
+            # Anyone who is watching commits in Pagure will most likely
+            # be interested in Bodhi updates on the package.
+            if 'commits' in watcher_levels:
+                watchers.add(watcher)
```

Comment:
I can't find any parts of Bodhi that use watchers. @ralphbean, do you think we can drop watchers? If so, I say let's just drop it. We can always add it later if it turns out we are wrong.